LEGO1/lego/legoomni/src/build/legocarbuildpresenter.cpp

```diff
@@ -218,6 +365,55 @@ void LegoCarBuildAnimPresenter::FUN_10079680(LegoChar* p_param)
 	// TODO
 }
 
+// FUNCTION: LEGO1 0x100796b0
+// FUNCTION: BETA10 0x10071f3c
+LegoAnimNodeData* LegoCarBuildAnimPresenter::FindNodeDataByName(LegoTreeNode* p_treeNode, const LegoChar* p_name)
+{
+	LegoAnimNodeData* data = NULL;
+
+	if (p_treeNode) {
+		data = (LegoAnimNodeData*) p_treeNode->GetData();
+
+		if (stricmp(data->GetName(), p_name) == 0) {
+			return data;
+		}
+
+		for (MxS32 i = 0; i < p_treeNode->GetNumChildren(); i++) {
+			data = FindNodeDataByName(p_treeNode->GetChildren()[i], p_name);
+
+			if (data) {
+				return data;
+			}
+		}
+	}
+	return NULL;
+}
+
+// FUNCTION: LEGO1 0x10079720
+// FUNCTION: BETA10 0x10071fec
+LegoTreeNode* LegoCarBuildAnimPresenter::FindNodeByName(LegoTreeNode* p_treeNode, const LegoChar* p_name)
+{
+	LegoAnimNodeData* data = NULL;
+	LegoTreeNode* node = NULL;
+
+	if (p_treeNode) {
+		data = (LegoAnimNodeData*) p_treeNode->GetData();
+
+		if (stricmp(data->GetName(), p_name) == 0) {
+			return p_treeNode;
+		}
+
+		for (MxS32 i = 0; i < p_treeNode->GetNumChildren(); i++) {
+			node = FindNodeByName(p_treeNode->GetChildren()[i], p_name);
+
+			if (node) {
+				return node;
+			}
+		}
+	}
```

Comment:
Needs blank line before return

Reply:
I was confused for a second because Github's preview is not helpful at all here, suggesting that a different `return` statement was meant. Fixed it